extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -9,203 +12,306 @@ class SpecialDiscussionsLogController extends WikiaSpecialPageController {
+		} else {
+			$searchQuery = null;
+		}
+
+		$queryStringParams = [ ];
+		if ( $searchQuery ) {
+			$queryStringParams[ $searchQuery->getKeyName() ] = $searchQuery->getKey();
+		}
+		$this->setVal( 'inputForm', $this->sendSelfRequest( 'inputForm', $queryStringParams ) );
+
+		if ( $searchQuery ) {
+			$this->requestUserLog( $searchQuery );
 		}
 	}
 
+	/**
+	 * Returns an array containing userName and ipAddress
+	 * @return array
+	 */
+	private function readAndValidateInput() {
+		$userName = $this->getVal( UserQuery::getKeyName(), null );
+		$ipAddress = $this->getVal( IpAddressQuery::getKeyName(), null );
+
+		if ( !empty( $userName ) && !empty( $ipAddress ) ) {
+			throw new \InvalidArgumentException( wfMessage( 'discussionslog-multiple-input-error' )->escaped() );
+		}
+
+		return [ $userName, $ipAddress ];
+	}
+
+	private function requestUserLog( SearchQuery $searchQuery ) {
+		$this->setVal(
+			'userLog',
+			$this->sendSelfRequest( 'userLog', [ $searchQuery->getKeyName() => $searchQuery->getKey() ] ) );
+	}
+
 	public function inputForm() {
-		$this->userName = $this->getVal( 'username' );
-		$this->userNameLabel = wfMessage( 'discussionslog-username-label' )->escaped();
-		$this->viewLogsAction = wfMessage( 'discussionslog-view-logs' )->escaped();
+		list( $userName, $ipAddress ) = $this->readAndValidateInput();
+
+		$this->response->setValues( [
+			'userName' => $userName,
+			'ipAddress' => $ipAddress,
+			'userNameLabel' => wfMessage( 'discussionslog-username-label' )->escaped(),
+			'ipAddressLabel' => wfMessage( 'discussionslog-ip-address-header' )->escaped(),
+			'orLabel' => wfMessage( 'discussionslog-or-label' )->escaped(),
+			'viewLogsAction' => wfMessage( 'discussionslog-view-logs' )->escaped(),
+		] );
 	}
 
 	private function constructKibanaUrl( $dayOffset ) {
 		global $wgConsulUrl;
 
-		$esUrl = (new  Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, 'query', 'sjc' ))->getUrl( 'es' );
+		$esUrl = (new Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, 'query', 'sjc' ))->getUrl( 'es' );
 		$date = time() - ( $dayOffset * 24 * 60 * 60 );
 
 		return 'http://' . $esUrl . '/logstash-' . date( 'Y.m.d', $date ) . '/_search';
 	}
 
-	private function getUserIdByUsername( $userName ) {
+	private function getUserByUsername( $userName ) {
 		if ( empty( $userName ) ) {
-			throw new InvalidArgumentException( self::NO_USER_MATCH_ERROR );
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
 		}
 
 		$user = User::newFromName( $userName );
-		if ( !$user || empty( $user->getId() ) ) {
-			throw new InvalidArgumentException( self::NO_USER_MATCH_ERROR );
+		if ( !$user ) {
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
+		}
+
+		return $user;
+	}
+
+	private function getUserById( $userId ) {
+		if ( !$userId ) {
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
```

Comment:
What do you think about not localizing these error methods that only go to our logs? L164 looks like the only place this message bubbles up to the user, and it should definitely be translated there, but for the rest of these it'll make searching for errors harder since we'll have to do it in multiple languages.